plus/src/test/java/org/apache/calcite/slt/SqlLogicTests.java

```diff
@@ -0,0 +1,337 @@
+        writer.write(this.toString());
+      }
+    }
+
+    /**
+     * True if there is an entry for the specified test file.
+     *
+     * @param test Test file name.
+     */
+    public boolean contains(String test) {
+      return this.testResults.containsKey(test);
+    }
+  }
+
+  /**
+   * Summaries produced for the current run.
+   * Must be static since it is written by the `findRegressions`
+   * static method.
+   */
+  static AllTestSummaries testSummaries = new AllTestSummaries();
+
+  static final String GOLDEN_FILE = "/slttestfailures.txt";
+  /**
+   * Summaries checked-in as resources that we compare against.
+   */
+  static AllTestSummaries goldenTestSummaries =
+      new AllTestSummaries()
+          .read(SqlLogicTests.class.getResourceAsStream(GOLDEN_FILE));
+
+  private static TestStatistics launchSqlLogicTest(String... args) throws IOException {
+    OptionsParser options = new OptionsParser(false, System.out, System.err);
+    CalciteExecutor.register(options);
+    return net.hydromatic.sqllogictest.Main.execute(options, args);
+  }
+
+  TestSummary shortSummary(String file, TestStatistics statistics) {
+    return new TestSummary(file, statistics.getFailedTestCount());
+  }
+
+  /**
+   * The following tests currently timeout during execution.
+   * Technically these are Calcite bugs.
+   */
+  Set<String> timeout =
+      ImmutableSet.of("test/select5.test",
+          "test/random/groupby/slt_good_10.test");
+
+  /**
+   * The following tests contain SQL statements that are not supported by HSQLDB.
+   */
+  Set<String> unsupported =
+      ImmutableSet.of("test/evidence/slt_lang_replace.test",
+          "test/evidence/slt_lang_createtrigger.test",
+          "test/evidence/slt_lang_droptrigger.test",
+          "test/evidence/slt_lang_update.test",
+          "test/evidence/slt_lang_reindex.test");
+
+  void runOneTestFile(String testFile) throws IOException {
+    if (timeout.contains(testFile)) {
+      return;
+    }
+    if (unsupported.contains(testFile)) {
+      return;
+    }
+
+    // The arguments below are command-line arguments for the sql-logic-test
+    // executable from the hydromatic project.  The verbosity of the
+    // output can be increased by adding more "-v" flags to the command-line.
+    // By increasing verbosity even more you can get in the output a complete stack trace
+    // for each error caused by an exception.
+    TestStatistics res = launchSqlLogicTest("-v", "-e", "calcite", testFile);
+    assertThat(res, notNullValue());
+    assertThat(res.getParseFailureCount(), is(0));
+    assertThat(res.getIgnoredTestCount(), is(0));
+    assertThat(res.getTestFileCount(), is(1));
+    res.printStatistics(System.err);  // Print errors found
+    TestSummary summary = this.shortSummary(testFile, res);
+    boolean regression = goldenTestSummaries.regression(summary);
+    Assumptions.assumeFalse(regression, "Regression in " + summary.file);
```

Comment:
I thought we were always checking the results against the golden file; if that's the case then why do we have the assertThat statements? Do we need both assertThat and assumeFalse, if yes why?

Reply:
OK I understood, the assertThat are basically to ensure that the file was processes correctly.